pkg/types/baremetal/validation/platform_test.go

```diff
@@ -740,3 +272,169 @@ func TestValidatePlatform(t *testing.T) {
 		})
 	}
 }
+
+///////////////////////////////////////////////////////////////////////////////
+
+type hostBuilder struct {
```

Comment:
this process is fine, but validations in the other parts of the codebase solve this using modifiers like https://github.com/openshift/installer/blob/c09a61047a726d069664cedfef9f6c6e3a896eec/pkg/asset/installconfig/aws/validation_test.go#L133-L142

I'm fine with all of these, but since these builders are not generated, we will end up adding functions for each new field, and most of the times the modifications to the valid examples happens in batches (multiple fields together)

Reply:
Potentially the builder could be generated but I wanted to keep it simple, since function for a new field is added only once and very quickly, and after that can be reused by many developers when writing tests.
About batch modifications, not sure If I understood correctly: fluent builders allows modifying multiple fields together through the usage of method chaining (see for example https://github.com/andfasano/installer/blob/ab40687d1715529356df8c22d2b8b0abd3504db0/pkg/types/baremetal/validation/platform_test.go#L54)